# Exploring sentence-transformers library 

The sentence-transformers library can be installed using pip as shown in the following code:


In [1]:
%%capture
!pip install sentence_transformers==0.4.0


The researchers of Sentence-BERT have also made their pre-trained Sentence-BERT models available online. All available pre-trained models can be found here (https://public.ukp.informatik.tu-darmstadt.de/reimers/sentence-transformers/v0.2/).

We can find pre-trained models named bert-base-nli-cls-token, bert-base-nli-mean-token, roberta-base-nli-max-tokens, distilbert-base-nli-mean-tokens, and so on. Let us understand what this means. 

- bert-base-nli-cls-token is a pre-trained Sentence-BERT model where we have taken a pre-trained BERT-base model and fine-tuned it with the NLI dataset and the model uses [CLS] token as the sentence representation. 
- bert-base-nli-mean-token is a pre-trained Sentence-BERT model where we have taken a pre-trained BERT-base model and fine-tuned it with the NLI dataset and the model uses mean pooling strategy for computing sentence representation.  
- roberta-base-nli-max-tokens is a pre-trained Sentence-BERT model where we have taken a pre-trained RoBERTa-base model and fine-tuned it with the NLI dataset and the model uses max pooling strategy for computing sentence representation. 
- distilbert-base-nli-mean-token is a pre-trained Sentence-BERT model where we have taken a pre-trained DistilBERT-base model and fine-tuned it with the NLI dataset and the model uses mean pooling strategy for computing sentence representation. 

Thus, when we say the pre-trained Sentence-BERT model, it basically implies that we have taken a pre-trained BERT model and finetuned it using the Siamese/triplet network architecture. In the upcoming sections, we will learn how to use the pre-trained Sentence-BERT models. 


## Computing sentence representation using Sentence-BERT 
Let us see how to compute sentence representation using pre-trained Sentence-BERT. First, let us import the SentenceTransformer module from our sentence_transformers library. Use the following command:



In [2]:
from sentence_transformers import SentenceTransformer

Download and load the pre-trained Sentence-BERT:

In [3]:
model = SentenceTransformer('bert-base-nli-mean-tokens')


Define the sentence to which we need to compute the sentence representation:

In [4]:
sentence = 'paris is a beautiful city'


Compute the sentence representation using our pre-trained Sentence-BERT model with encode function: 

In [5]:
sentence_representation = model.encode(sentence)


Now, let us check the shape of our representation:

In [6]:
print(sentence_representation.shape)

(768,)



As we can see our sentence representation size is 768. In this way, we can use the pre-trained Sentence-BERT model and obtain the fixed-length sentence representation. 